### ISSUE TITLE
Expression: "Thrown during type checking", "System.NullReferenceException"

### ISSUE THREAD

**Issue by FelesNoctis** [joelday/papyrus-lang#90]
**Describe the bug**
Error log found here: https://pastebin.com/y81gSVYP
When the extension is attempting read and process an expression, any time the expression is incomplete it will throw this error. It appears to "jump the gun", trying to evaluate/cast empty space.

**To Reproduce**

1. Create an expression.
`debug.Notification("Old Value: ")`
2. Edit the expression with additional information.
`debug.Notification("Old Value: " +)`
3. Error appears at this point while typing.

**Comment by minkir014** [comment]
I think this is ideal. This happens to all languages for me. When you leave a plus sign without something after it, it gives an error but the error disappears when I complete something after it

**Comment by FelesNoctis** [comment]
What languages are you looking at? Generally it would be the job of a linter to report an error in the code, not an error breaking the extension itself.

For instance, the same expression error in Javascript:
![](https://i.imgur.com/QvvtFvG.png)

You do already have a [linter](https://i.imgur.com/2VHyne6.png) for Papyrus-Lang, shouldn't the error be funneling there similar to everything else?

**Comment by minkir014** [comment]
I didn't know it breaks the extension itself.

**Comment by FelesNoctis** [comment]
Yeah. The error in question is due to the extension not properly handling the circumstance in question. The VSCode error log was posted to pastebin and linked above.

**Comment by joelday** [comment]
This will be addressed in the rewrite in progress.
